### PR TITLE
Deep Freeze Mock Messages

### PR DESCRIPTION
**Pull request by anthony-murphy**
We are seeing bugs where object references are being passed on ops which then leads to corruptions when those object are modified later. This manifests more strongly in out tests, where the object survive the op processing round trip, but this could still cause real problems on a local client. This change add a deep freeze on the object passed into the op processing pipeline such that we can catch if any of the referenced objects are later modified.

### CODE REVIEW

File: packages/runtime/test-runtime-utils/src/mocks.ts
Comment (anthony-murphy) [inline packages/runtime/test-runtime-utils/src/mocks.ts:526]: everything else is just moves. this the only change

**Comment by anthony-murphy** [comment]
@jzaffiro and @Abe27342 I'm waiting for #20620 before i merge this, as i don't want to block that, but open to feedback before then

File: packages/dds/sequence/src/intervalCollection.ts
Comment (anthony-murphy) [inline packages/dds/sequence/src/intervalCollection.ts:1284]: found some test failures in interval collections that these changes fix

**Review by Abe27342 — APPROVED** [review]

**Pull request merged**
